src/wp-content/themes/twentyfifteen/js/view-transitions.js

```diff
@@ -0,0 +1,91 @@
+if ( !! window.navigation && 'CSSViewTransitionRule' in window ) {
+	const getParentElement = ( element, selector ) => {
+		element = element.parentElement;
+		while ( element && element !== document ) {
+			if ( ! selector || element.matches( selector ) ) {
+				return element;
+			}
+		}
+		return null;
+	};
+
+	const setTemporaryViewTransitionNames = async ( entries, vtPromise ) => {
+		for ( const [ element, name ] of entries ) {
+			if ( ! element ) {
+				continue;
+			}
+			element.style.viewTransitionName = name;
+		}
+
+		await vtPromise;
+
+		for ( const [ element, name ] of entries ) {
+			if ( ! element ) {
+				continue;
+			}
+			element.style.viewTransitionName = '';
+		}
+	};
+
+	window.addEventListener( 'pageswap', ( e ) => {
+		if ( e.viewTransition ) {
+			if ( document.body.classList.contains( 'single' ) ) {
+				const article = document.querySelectorAll( 'article.post' );
+				if ( article.length !== 1 ) {
+					return;
+				}
+
+				setTemporaryViewTransitionNames( [
+					[ article[ 0 ].querySelector( '.entry-title' ), 'post-title' ],
+					[ article[ 0 ].querySelector( '.post-thumbnail' ), 'post-thumbnail' ],
+				], e.viewTransition.finished );
+			} else if ( document.body.classList.contains( 'home' ) || document.body.classList.contains( 'archive' ) ) {
+				const articleLink = document.querySelector( 'article.post a[href="' + e.activation.entry.url + '"]' );
+				if ( ! articleLink ) {
+					return;
+				}
+
+				const article = getParentElement( articleLink, 'article.post' );
+
+				setTemporaryViewTransitionNames( [
+					[ article.querySelector( '.entry-title' ), 'post-title' ],
+					[ article.querySelector( '.post-thumbnail' ), 'post-thumbnail' ],
+				], e.viewTransition.finished );
+			}
+		}
+	} );
+
+	window.addEventListener( 'pagereveal', ( e ) => {
+		if ( ! window.navigation.activation.from ) {
+			return;
+		}
+
+		if ( e.viewTransition ) {
+			if ( document.body.classList.contains( 'single' ) ) {
+				const article = document.querySelectorAll( 'article.post' );
+				if ( article.length !== 1 ) {
+					return;
+				}
+
+				setTemporaryViewTransitionNames( [
+					[ article[ 0 ].querySelector( '.entry-title' ), 'post-title' ],
+					[ article[ 0 ].querySelector( '.post-thumbnail' ), 'post-thumbnail' ],
+				], e.viewTransition.ready );
+			} else if ( document.body.classList.contains( 'home' ) || document.body.classList.contains( 'archive' ) ) {
+				const articleLink = document.querySelector( 'article.post a[href="' + window.navigation.activation.from.url + '"]' );
+				if ( ! articleLink ) {
+					return;
+				}
+
+				const article = getParentElement( articleLink, 'article.post' );
+
+				setTemporaryViewTransitionNames( [
+					[ article.querySelector( '.entry-title' ), 'post-title' ],
+					[ article.querySelector( '.post-thumbnail' ), 'post-thumbnail' ],
+				], e.viewTransition.ready );
+			}
+		}
+	} );
```

Comment:
@noamr @bramus Maybe you can help me out here: With this current logic, whenever I click on a link to open a WordPress post that does not have a featured image (`post-thumbnail`) assigned, my browser gets stuck and seems to endlessly load the page.

Maybe it's just a simple logic flaw, but I'm not sure what's missing. My `setTemporaryViewTransitionNames` implementation has a condition to skip any elements that are `null` (via `! element`), so I would think such a scenario should be covered. But maybe something else is wrong.